Tests/test_DiscreteMath.py

```
@@ -0,0 +1,32 @@
+    assert compute_gcd(12, 36) == 12
+
+def test_compute_gcd_identical():
+    assert compute_gcd(7, 7) == 7
+    assert compute_gcd(100, 100) == 100
+
+def test_compute_gcd_common_factors():
+    assert compute_gcd(48, 18) == 6
+    assert compute_gcd(54, 24) == 6
+
+def test_compute_gcd_errors():
+    with pytest.raises(ValueError, match="Both numbers must be positive."):
```

Ruff (0.15.17)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)